### PR TITLE
add deploy mode for klusterlet

### PR DESCRIPTION
**Pull request by zhujian7**
Signed-off-by: zhujian <jiazhu@redhat.com>

### CODE REVIEW

**Comment by zhujian7** [comment]
Because of https://github.com/openshift/build-machinery-go/pull/56, the verity failed.

**Review by qiujian16 — COMMENTED** [review]
/approve
/assign @zhiweiyin318 @xuezhaojun

**Comment by zhujian7** [comment]
cc @ianzhang366

File: operator/v1/types.go [hunk 1]
```diff
@@ -50,29 +50,45 @@ type ClusterManagerSpec struct {
...
+	// For cluster-manager:
+	//   - In Default mode, the Hub is installed as a whole and all parts of Hub are deployed in the same cluster.
+	//   - In Detached mode, only crd and configurations are installed on one cluster(defined as hub-cluster). Controllers run in another cluster (defined as management-cluster) and connect to the hub with the kubeconfig in secret of "external-hub-kubeconfig"(a kubeconfig of hub-cluster with cluster-admin permission).
+	// For klusterlet:
+	//   - In Default mode, all klusterlet related resources are deployed on the managed cluster.
+	//   - In Detached mode, only crd and configurations are installed on the spoke/managed cluster. Controllers run in another cluster (defined as management-cluster) and connect to the mangaged cluster with the kubeconfig in secret of "external-managed-kubeconfig"(a kubeconfig of managed-cluster with cluster-admin permission).
```
Comment (ianzhang366) [inline operator/v1/types.go:65]: @zhujian7 how do we specify the configuration of management-cluster to controllers?

Comment (zhujian7) [reply operator/v1/types.go:65]: In fact for registration-agent and work-agent, the configuration here specifically refers to the RBAC configuration(in the managed cluster). Currently, we don't need a configuration resource(cm, in the management cluster) for the controllers(registration-agent and work-agent), all configs are reflected by the flags.

**Pull request merged**